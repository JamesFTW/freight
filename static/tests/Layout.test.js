import React from 'react';
import renderer from 'react-test-renderer';
import Layout from '../components/Layout.jsx';
import {shallow} from 'enzyme';
import ReactTestUtils from 'react-dom/test-utils';

describe('<Layout />', () => {
  it('should render <Layout />', () => {

  const appList = [{
    "environments": {
      "production": {
        "defaultRef": "master"
      },
      "staging": {
        "defaultRef": "HEAD"
      },
    },
    "id": "1",
    "name": "freight",
    "repository": "https://github.com/getsentry/freight.git"
  }]

  const children = {
  "history": {
    "listenBefore": {},
    "listen": {},
    "transitionTo": {},
    "push": {},
    "replace": {},
    "go": {},
    "goBack": {},
    "goForward": {},
    "createKey": {},
    "createPath": {},
    "createHref": {},
    "createLocation": {},
    "setState": {},
    "registerTransitionHook": {},
    "unregisterTransitionHook": {},
    "pushState": {},
    "replaceState": {},
    "isActive": {},
    "match": {},
    "listenBeforeLeavingRoute": {}
  },
  "location": {
    "pathname": "/",
    "search": "",
    "hash": "",
    "state": null,
    "action": "PUSH",
    "key": "m0qx3x",
    "basename": "/",
    "query": {},
    "$searchBase": {}
  },
  "params": {},
  "route": {
    "exact": true,
    "path": "/",
    "component": {},
    "indexRoute": {},
    "childRoutes": {}
  },
  "routeParams": {},
  "routes": [
    {},
    {}
  ]
}

const shallowRenderer = ReactTestUtils.createRenderer();
const result = shallowRenderer.render(<Layout children={children} />);
expect(result).toMatchSnapshot();
  })
})
