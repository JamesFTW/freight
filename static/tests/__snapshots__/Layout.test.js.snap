// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render <Layout /> 1`] = `
<div>
  <Indicators />
  <header>
    <div
      className="container"
    >
      <div
        className="pull-right"
      >
        <Link
          className="btn btn-sm btn-default"
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/deploy"
        >
          Deploy
        </Link>
      </div>
      <h1>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/"
        >
          Freight
        </Link>
      </h1>
    </div>
  </header>
  <div
    className="body"
  >
    <div
      className="container"
    >
      <Unknown
        appList={null}
      />
    </div>
  </div>
</div>
`;
